EdFi.Buzz.UI/src/Feature/AdminSurvey/index.tsx

```diff
@@ -1,14 +1,276 @@
+    outline: none !important;
+    border-color: var(--denim) !important;
+  }
+
+  & > img {
+    height: 18px;
+    width: 18px;
+    margin: 10px 10px 10px 10px;
+  }
+
+  & > input {
+    height: 100%;
+    padding: 2px 2px 2px 2px;
+    border: none;
+    position: relative;
+    box-sizing: border-box;
+    min-width: 18rem;
+
+    :focus {
+      outline: none !important;
+      outline-width: 0px;
+    }
+
+    ::placeholder,
+    ::-webkit-input-placeholder {
+      font-style: italic;
+    }
+
+    :-moz-placeholder {
+      font-style: italic;
+    }
+
+    ::-moz-placeholder {
+      font-style: italic;
+    }
+
+    :-ms-input-placeholder {
+      font-style: italic;
+    }
+  }
+`;
+
+export const AdminSurvey: FunctionComponent<AdminSurveyComponentProps> = (props: AdminSurveyComponentProps) => {
   document.title = 'EdFi Buzz: Admin Surveys';
 
-  return <div>AdminSurveys {props.title ?? ''}</div>;
+  const [surveyFilteredList, setsurveyFilteredList] = useState([] as SurveyStatus[]);
+  const [surveyList, setsurveyList] = useState([] as SurveyStatus[]);
+  const [surveyToDelete, setsurveyToDelete] = useState(null as number);
+  const [searchText, setsearchText] = useState('' as string);
+  const surveyFilterRef = React.createRef<HTMLInputElement>();
+
+  if (!surveyFilteredList || surveyFilteredList.length === 0) {
+    props.api.survey.getSurveyStatus(
+      props.api.authentication.currentUserValue.teacher.staffkey, null)
+      .then((surveysValue) => {
+        setsurveyList(surveysValue);
+        setsurveyFilteredList(surveysValue);
+      });
+  }
+
+  function handleChange() {
+    setsearchText(surveyFilterRef.current.value);
+    if (searchText && searchText.length > 0) {
+      const upperSearchText = searchText.toUpperCase();
+      const filteredList = surveyList
+        .filter(s => (s.resultSummaryObj.survey.title as string).toUpperCase().includes(upperSearchText));
+      setsurveyFilteredList(filteredList);
+    } else {
+      setsurveyFilteredList(surveyList);
+    }
+  }
+
+  function setSurveyToDelete(surveyKey: number) {
+    setsurveyToDelete(surveyKey);
+  }
+
+  function deleteSurvey() {
+    if (surveyToDelete) {
+      props.api.survey
+        .deleteSurvey(props.api.authentication.currentUserValue.teacher.staffkey, surveyToDelete)
+        .then(() => {
+          const idx = surveyList.findIndex(el => el.surveykey === surveyToDelete);
+          if (idx > -1) {
+            surveyList.splice(idx, 1);
+          }
+          if (searchText) {
+            const textIndex = surveyFilteredList.findIndex(el => el.surveykey === surveyToDelete);
+            if (textIndex > -1) {
+              surveyFilteredList.splice(textIndex, 1);
+            }
+          }
+          setsurveyToDelete(null);
+        });
+    }
+  }
+
+  return (
+    <>
+      <MainContainer role='main' className='container'>
+        <HeadlineContainer>
+          <TitleSpanContainer>Administrate surveys</TitleSpanContainer>
```

Comment:
Can we change Administrate to Administer? That is more common. I checked with David already and the change is OK.

Reply:
Fix